Framework/Algorithms/src/RebinToWorkspace.cpp

```diff
@@ -43,55 +47,127 @@ void RebinToWorkspace::init() {
                   "Workspace2D histogram.");
 }
 
+namespace {
+bool needToRebin(const MatrixWorkspace_sptr &left,
+                 const MatrixWorkspace_sptr &rght) {
+  // if pointers match they are the same object
+  if (left == rght)
+    return false;
+
+  // see if there is the same number of histograms
+  const size_t numHist = left->getNumberHistograms();
+  if (numHist != rght->getNumberHistograms())
+    return true;
+
+  // look for first non-equal x-axis between the workspaces
+  for (size_t i = 0; i < numHist; ++i) {
+    if (left->getSpectrum(i).x() != rght->getSpectrum(i).x()) {
+      return true;
+    }
+  }
+  // everything must be the same
```

Comment:
Shouldn't you also check if you need to preserve events? If not, you always need to rebin in the case of `EventWorkspace`.

Reply:
Right, I missed that extra call to `createOutputWorkspace`. :+1: